network/network.go

```diff
@@ -3,17 +3,76 @@ package network
 import (
 	"crypto/ecdsa"
 	"fmt"
+	"io"
 	"net"
 
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/p2p/enode"
 	"github.com/ethereum/go-ethereum/p2p/enr"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethersphere/swarm/network/capability"
 )
 
 // BzzAddr implements the PeerAddr interface
 type BzzAddr struct {
-	OAddr []byte
-	UAddr []byte
+	OAddr        []byte
+	UAddr        []byte
+	Capabilities *capability.Capabilities
+}
+
+// EncodeRLP implements rlp.Encoder
```

Comment:
it should be enough to implement custom rlp for Capabilities within that package no?

Reply:
I thought so too, but it's not